daemon/oci_windows.go

```diff
@@ -288,6 +288,23 @@ func (daemon *Daemon) createSpecWindowsFields(c *container.Container, s *specs.S
 				if cs, err = readCredentialSpecRegistry(c.ID, csValue); err != nil {
 					return err
 				}
+			} else if match, csValue = getCredentialSpec("config://", splitsOpt[1]); match {
+				if csValue == "" {
+					return fmt.Errorf("no value supplied for config:// credential spec security option")
+				}
+
+				// if the container does not have a DependencyStore, then we
+				// return an error
+				if c.DependencyStore == nil {
```

Comment:
Wondering (reading from my phone so have limited sight in the whole code path) if we can catch this situation during _create_ of the container instead of here (as this is (I think) only ran at the last point, at container start)

eg somewhere in this code path (would have to check if that's the right place though) https://github.com/moby/moby/blob/e7a9a7cdbc8fc87cb2bf07b770bd8bfe52ab0959/daemon/create.go#L25

/cc @cpuguy83